Automattic/remote-data-blocks
# add google sheets field selection in settings

# Description

## Enhancements

- Extracted field selection input to common `FieldsSelection` component which can be reused for other data sources (Google Sheets and Airtable for now) (299a7a5)
- Added custom help text for field selection input to show hints and custom messages (61431f4)
- Added field selection input for Google Sheets data source. This doesn't yet auto-register blocks for this data source like it happends for Airtable. That would be added in a separate PR. (2b51404)

    | <img width="964" alt="Screenshot 2024-12-27 at 12 36 45 AM" src="https://github.com/user-attachments/assets/b52be909-fc93-41b4-994b-b16b2779f3b2" /> |
    | :---: |
    | Google Sheets settings form with field selection input |

## Bugfixes

- Fixed failure to generate token due to incorrectly parsed credentials input to and from Object to JSON string (bf99106)
- Fixed form state management for Google Sheets data source add/edit form which prevented creating/editing Google Sheets data source (bf99106)

# Testing

- Before checking out this branch, delete all the existing Google Sheets data sources configured using the plugin settings page. This is required as the new storage schema for Google Sheets Service Config has breaking changes.
- Start the local development environment ([docs](https://github.com/Automattic/remote-data-blocks/blob/trunk/docs/local-development.md))
- Try adding a new Google Sheets data source using the plugin settings page and verify that:
    - It works without any bugs and that you are able to select fields for the selected sheet.
    - When the fields are selected and you change the spreadsheet, the `Sheet` and `Fields` inputs are reset.
    - When the fields are selected and you change the sheet, the `Fields` input is reset.
    - If the sheet input is not selected, the `Fields` input is disabled.
    - When the sheet input is selected, the help text for the `Fields` input shows the message `Fetching fields...` until the fields are fetched. After the fields are fetched, the input should be enabled and the help text should change to hint regarding the `Fields` input.
